astropy/io/fits/connect.py

```diff
@@ -424,10 +426,14 @@ def write_table_fits(input, output, overwrite=False):
     if isinstance(output, str) and os.path.exists(output):
         if overwrite:
             os.remove(output)
-        else:
+        elif not append:
             raise OSError(f"File exists: {output}")
```

Comment:
Not sure if we should allow `(overwrite=True, append=True)`. It seems a weird combination of parameters.

Reply:
It does seem a bit weird, but I don't think the behaviour is in any way surprising, and it's not clear to me that there's any benefit in explicitly forbidding it. It could be that there's a use case where it's handy, although I can't think of one. The hdf5 version also has both parameters, although the behaviour is a bit different.

Reply:
Yeah, I think it's fine to allow.  It's one of those cases where "there's probably not a strong reason to do it intentionally" but the behavior is at least consistent and unsurprising.

Reply:
Yeah, ok, works for me.